File: examples/Pendulum/PendulumState.m
classdef PendulumState < LCMCoordinateFrame & Singleton
  
  methods
    function obj=PendulumState()
      obj = obj@LCMCoordinateFrame('PendulumState','drake.examples.Pendulum.lcmt_pendulum_x','x',{'theta','thetadot'});
      obj = obj@Singleton();
    end
  end
end

File: examples/pendulum/pendulum_state.cc
#include "drake/examples/pendulum/pendulum_state.h"

namespace drake {
namespace examples {
namespace pendulum {

const int PendulumStateIndices::kNumCoordinates;
const int PendulumStateIndices::kTheta;
const int PendulumStateIndices::kThetadot;

const std::vector<std::string>& PendulumStateIndices::GetCoordinateNames() {
  static const drake::never_destroyed<std::vector<std::string>> coordinates(
      std::vector<std::string>{
          "theta",
          "thetadot",
      });
  return coordinates.access();
}

}  // namespace pendulum
}  // namespace examples
}  // namespace drake
